### PR TITLE
Making docs better

### PR DESCRIPTION
**Pull request by muskan3006**
References #27792
This pull request is regarding remove some non-helpful phrases in the doc. I did changes in four files under streams.

### CODE REVIEW

File: akka-docs/src/main/paradox/stream/operators/ActorFlow/ask.md [hunk 1]
```diff
@@ -28,16 +28,15 @@ If any of the asks times out it will fail the stream with an @apidoc[AskTimeoutE
...
+* a `makeMessage` function to create the message sent to the actor from the incoming element, and the actor ref accepting the actor's reply message and a timeout.
 
 See also:
 
 * @ref[Flow.ask](../Source-or-Flow/ask.md) for the classic actors variant
 
 ## Examples
 
-The `ActorFlow.ask` sends a message to the actor which expects `Asking` messages which contain the actor ref for replies of type `Reply`. The replies are emitted when received and the `map` extracts the message `String`.
+The `ActorFlow.ask` sends a message to the actor which expects `Asking` messages which contain the actor ref for replies of type `Reply`. When it receives a reply, it emits it  and the `map` extracts the message `String`.
```
Comment (raboof) [inline akka-docs/src/main/paradox/stream/operators/ActorFlow/ask.md:39]: I'm not sure it is very clear what "it" is here. Perhaps:

```suggestion
The `ActorFlow.ask` sends a message to the actor which expects `Asking` messages which contain the actor ref for replies of type `Reply`. The replies are emitted by the `askFlow` and the `map` extracts the message `String`.
```

File: akka-docs/src/main/paradox/stream/operators/ActorFlow/ask.md [hunk 2]
```diff
@@ -52,11 +51,11 @@ Java
 
 **emits** when the futures (in submission order) created by the ask pattern internally are completed
 
-**backpressures** when the number of futures reaches the configured parallelism and the downstream backpressures
+**backpressure** when the number of futures reaches the configured parallelism and the downstream backpressure
```
Comment (raboof) [inline akka-docs/src/main/paradox/stream/operators/ActorFlow/ask.md:54]: 'backpressure' is intended to be a verb here, so it needs to be 'backpressures' in both cases.

File: akka-docs/src/main/paradox/stream/operators/ActorFlow/ask.md [hunk 3]
```diff
@@ -52,11 +51,11 @@ Java
...
 
 **completes** when upstream completes and all futures have been completed and all elements have been emitted
 
-**fails** when the passed in actor terminates, or a timeout is exceeded in any of the asks performed
+**fails** when the passed in an actor terminates, or a timeout exceeds in any of the asks performed
```
Comment (raboof) [inline akka-docs/src/main/paradox/stream/operators/ActorFlow/ask.md:58]: 'the passed in actor' is the subject of this sentence, inserting 'an' here makes it invalid.

While changing the passive 'is exceeded' into the active 'exceeds' is probably nicer, the sentence needs to be rearranged further. Perhaps:

```suggestion
**fails** when the passed-in actor terminates, or when any of the `ask`s exceed a timeout
```

File: akka-docs/src/main/paradox/stream/operators/ActorSink/actorRef.md [hunk 4]
```diff
@@ -49,6 +49,6 @@ See also:
 
 **cancels** when the actor terminates
 
-**backpressures** never
+**backpressure** never
```
Comment (raboof) [inline akka-docs/src/main/paradox/stream/operators/ActorSink/actorRef.md:52]: these are all verbs

```suggestion
**backpressures** never
```

File: akka-docs/src/main/paradox/stream/operators/ActorSink/actorRefWithBackpressure.md [hunk 5]
```diff
@@ -1,6 +1,6 @@
 # ActorSink.actorRefWithBackpressure
 
-Sends the elements of the stream to the given @java[`ActorRef<T>`]@scala[`ActorRef[T]`] of the new actors API with backpressure, to be able to signal demand when the actor is ready to receive more elements.
+Sends the elements of the stream to the given @java[`ActorRef<T>`]@scala[`ActorRef[T]`] of the new actors API with the backpressure, to be able to signal demand when the actor is ready to receive more elements.
```
Comment (raboof) [inline akka-docs/src/main/paradox/stream/operators/ActorSink/actorRefWithBackpressure.md:3]: This does not make sense to me.

```suggestion
Sends the elements of the stream to the given @java[`ActorRef<T>`]@scala[`ActorRef[T]`] of the new actors API with backpressure, to be able to signal demand when the actor is ready to receive more elements.
```

File: akka-docs/src/main/paradox/stream/operators/ActorSink/actorRefWithBackpressure.md [hunk 6]
```diff
@@ -44,7 +44,7 @@ Java
 
 **cancels** when the actor terminates
 
-**backpressures** when the actor acknowledgement has not arrived
+**backpressure** when the actor acknowledgement has not arrived
```
Comment (raboof) [inline akka-docs/src/main/paradox/stream/operators/ActorSink/actorRefWithBackpressure.md:47]:
```suggestion
**backpressures** when the actor acknowledgement has not arrived
```

File: akka-docs/src/main/paradox/stream/operators/ActorSource/actorRefWithBackpressure.md [hunk 7]
```diff
@@ -23,7 +23,7 @@ This operator is included in:
 ## Description
 
 Materialize an @java[`ActorRef<T>`]@scala[`ActorRef[T]`], sending messages to it will emit them on the stream. The actor responds with the provided ack message
-once the element could be emitted allowing for backpressure from the source. Sending another message before the previous one has been acknowledged will fail the stream.
+once the element could be emitted allowing for the backpressure from the source. Sending another message before the previous one has been acknowledged will fail the stream.
```
Comment (raboof) [inline akka-docs/src/main/paradox/stream/operators/ActorSource/actorRefWithBackpressure.md:26]:
```suggestion
once the element could be emitted allowing for backpressure from the source. Sending another message before the previous one has been acknowledged will fail the stream.
```

File: akka-docs/src/main/paradox/stream/operators/ActorSource/actorRefWithBackpressure.md [hunk 8]
```diff
@@ -53,7 +53,7 @@ Java
 
 @@@div { .callout }
 
-**emits** when there is demand and a message is sent to the materialized @scala[`ActorRef[T]`]@java[`ActorRef<T>`]
+**emits** when there is demand, and a message is sent to the materialized @scala[`ActorRef[T]`]@java[`ActorRef<T>`]
```
Comment (raboof) [inline akka-docs/src/main/paradox/stream/operators/ActorSource/actorRefWithBackpressure.md:56]: I think we can improve this further:
```suggestion
**emits** when a message is sent to the materialized @scala[`ActorRef[T]`]@java[`ActorRef<T>`] it is emitted as soon as there is demand from downstream
```

**Review by raboof — COMMENTED** [review]
Thanks! I've added some comments and proposals for further improvements.

File: akka-docs/src/main/paradox/stream/operators/ActorFlow/ask.md (hunk 1 above)
Comment (ignasi35) [reply akka-docs/src/main/paradox/stream/operators/ActorFlow/ask.md:39]: Maybe we can even remove the double-`which` and some pronouns:
```suggestion
The `ActorFlow.ask` sends a message to the actor. The actor expects `Asking` messages which contain the actor ref for replies of type `Reply`. When the actor for replies receives a reply, the `ActorFlow.ask` stream stage emits the reply and the `map` extracts the message `String`.
```

File: akka-docs/src/main/paradox/stream/operators/ActorSource/actorRefWithBackpressure.md [hunk 9]
```diff
@@ -34,7 +34,7 @@ See also:
 
 ## Example
 
-With `actorRefWithBackpressure` two actors get into play: one which feeds the stream and is materialized when the stream runs, and one which is provided by us and gets the ack signal when an element is emitted into the stream.
+With `actorRefWithBackpressure` two actors get into play: one which feeds the stream and is materialized when the stream runs, and one which is provided by us which gets the ack signal when an element is emitted into the stream.
```
Comment (ignasi35) [inline akka-docs/src/main/paradox/stream/operators/ActorSource/actorRefWithBackpressure.md:37]: Maybe fewer `which`s and even a numbered list could be helpful:

```suggestion
With `actorRefWithBackpressure` two actors get into play: 

1. An actor that is materialized when the stream runs. It feeds the stream.
2. An actor provided by the user. It gets the ack signal when an element is emitted into the stream.
```

In the list I also reordered the sentences to the "who provides the actor. What does the actor do" format.

**Review by ignasi35 — CHANGES_REQUESTED** [review]
Thanks @muskan3006 for all these improvement suggestions. Keeping the docs for these (very common) operators clean and easy to understand is very valuable.

I left a couple of comments next to @raboof's. :-)

File: akka-docs/src/main/paradox/stream/operators/ActorSource/actorRefWithBackpressure.md (hunk 9 above)
Comment (muskan3006) [reply akka-docs/src/main/paradox/stream/operators/ActorSource/actorRefWithBackpressure.md:37]: This one is good. I'll do this.

File: akka-docs/src/main/paradox/stream/operators/ActorFlow/ask.md [hunk 10]
```diff
@@ -28,16 +28,16 @@ If any of the asks times out it will fail the stream with an @apidoc[AskTimeoutE
...
+* a timeout.
 
 See also:
 
 * @ref[Flow.ask](../Source-or-Flow/ask.md) for the classic actors variant
 
 ## Examples
 
-The `ActorFlow.ask` sends a message to the actor which expects `Asking` messages which contain the actor ref for replies of type `Reply`. The replies are emitted when received and the `map` extracts the message `String`.
+The `ActorFlow.ask` sends a message to the actor. The actor expects `Asking` messages which contain the actor ref for replies of type `Reply`. When the actor for replies receives a reply, the `ActorFlow.ask` stream stage emits the reply and the `map` extracts the message `String`.
```
Comment (raboof) [inline akka-docs/src/main/paradox/stream/operators/ActorFlow/ask.md:40]: Nice!

File: akka-docs/src/main/paradox/stream/operators/ActorSink/actorRefWithBackpressure.md [hunk 11]
```diff
@@ -4,7 +4,7 @@ Sends the elements of the stream to the given @java[`ActorRef<T>`]@scala[`ActorR
 
 @ref[Actor interop operators](../index.md#actor-interop-operators)
 
-## Dependency
+## Dependency
```
Comment (raboof) [inline akka-docs/src/main/paradox/stream/operators/ActorSink/actorRefWithBackpressure.md:7]:
```suggestion
## Dependency
```

**Review by raboof — APPROVED** [review]

File: akka-docs/src/main/paradox/stream/operators/ActorFlow/ask.md (hunk 10 above)
Comment (muskan3006) [reply akka-docs/src/main/paradox/stream/operators/ActorFlow/ask.md:40]: Thanks

File: akka-docs/src/main/paradox/stream/operators/ActorSource/actorRefWithBackpressure.md [hunk 12]
```diff
@@ -34,7 +34,9 @@ See also:
 
 ## Example
 
-With `actorRefWithBackpressure` two actors get into play: one which feeds the stream and is materialized when the stream runs, and one which is provided by us and gets the ack signal when an element is emitted into the stream.
+With `actorRefWithBackpressure` two actors get into play: 
+1. An actor that is materialized when the stream runs. It feeds the stream.
```
Comment (patriknw) [inline akka-docs/src/main/paradox/stream/operators/ActorSource/actorRefWithBackpressure.md:38]: I think there must be an empty line before the list

**Review by patriknw — COMMENTED** [review]
thanks, one small thing

File: akka-docs/src/main/paradox/stream/operators/ActorSource/actorRefWithBackpressure.md (hunk 12 above)
Comment (muskan3006) [reply akka-docs/src/main/paradox/stream/operators/ActorSource/actorRefWithBackpressure.md:39]: Done

**Review by ignasi35 — APPROVED** [review]
hi @muskan3006, apologies for the delay, I missed the notifications on this. 
It looks great, thanks!

**Comment by muskan3006** [comment]
> hi @muskan3006, apologies for the delay, I missed the notifications on this.
> It looks great, thanks!

It's okay, thanks for approving

**Pull request merged**